src/Microsoft.AspNetCore.Http.Connections/Internal/HttpConnectionDispatcher.cs

```diff
@@ -630,21 +627,49 @@ private async Task<bool> EnsureConnectionStateAsync(HttpConnectionContext connec
                 {
                     // Set the request trace identifier to the current http request handling the poll
                     existing.TraceIdentifier = context.TraceIdentifier;
-                    existing.User = context.User;
+
+                    // Don't copy the identity if it's a windows identity
+                    // We specifically clone the identity on first poll if it's a windows identity
+                    // There would be a race here if we swapped the new User and the application tried to access the User
+                    if (context.User.Identity.GetType() != typeof(WindowsIdentity))
```

Comment:
use is

Reply:
Yeah, I was kinda wondering why you used `GetType`

Reply:
Other languages crowding my head :/

Reply:
```
if(context.User.Identity.getType() instanceof WindowsIdentity.class)
```